packages/docs/src/mdx/colorSelector/ColorPicker.tsx

```diff
@@ -0,0 +1,170 @@
+  ColorName,
+  CSSColor,
+  spacing,
+  componentSize,
+  PopOver,
+} from 'practical-react-components-core'
+import { useClickOutside } from 'react-hooks-shareable'
+
+interface ColorBoxProps {
+  readonly color?: string
+  readonly selected: boolean
+}
+const ColorBox = styled.div<ColorBoxProps>`
+  background-color: ${({ color }) => color};
+  cursor: pointer;
+  height: ${componentSize.small};
+  width: ${componentSize.small};
+  ${({ selected }) =>
+    selected
+      ? `height: ${componentSize.mini};
+  width: ${componentSize.mini};
+  margin:2px`
+      : '0px'};
+`
+const SelectedColorBox = styled.div<ColorBoxProps>`
+  ${({ selected }) => (selected ? `border:2px solid;` : 'none')};
+`
+
+const ColorSegment = styled.div`
+  background-color: ${({ color }) => color};
+  cursor: pointer;
+  height: 20px;
+  border-radius: ${shape.radius.medium};
+  margin: 0 ${spacing.medium};
+`
+interface PaletteContainerProps {
+  readonly changeWidth?: number
+}
+
+const PaletteContainer = styled.div<PaletteContainerProps>`
+  overflow-y: scroll;
+  background-color: ${({ theme }) => theme.color.background()};
+  border: ${({ theme }) => theme.color.element13()} 2px solid;
+  border-radius: ${shape.radius.medium};
+  height: 300px;
+  display: flex;
+  flex-wrap: wrap;
+  width: ${({ changeWidth }) => changeWidth}px;
+  min-width: 115px;
+`
+interface PaletteColorProps {
+  readonly colorName: ColorName
+  readonly colorValue: CSSColor
+  readonly onChosenColor: (colorName: ColorName) => void
+  readonly selected: boolean
+}
+
+const PaletteColor: React.VFC<PaletteColorProps> = ({
+  colorName,
+  colorValue,
+  onChosenColor,
+  selected,
+}) => {
+  const onClick = useCallback(() => {
+    onChosenColor(colorName)
+  }, [colorName, onChosenColor])
+  return (
+    <div key={colorName}>
+      <Tooltip text={colorName}>
+        <SelectedColorBox selected={selected}>
+          <ColorBox
+            selected={selected}
+            color={colorValue()}
+            onClick={onClick}
+          />
+        </SelectedColorBox>
+      </Tooltip>
+    </div>
+  )
+}
+
+export interface ColorPickerProps {
+  readonly onChange: (value: ColorName) => void
+  readonly value: ColorName
+}
+
+export const ColorPicker: React.VFC<ColorPickerProps> = ({
+  onChange,
+  value,
+}) => {
+  const [directionUp, setDirectionUp] = useState(false)
+  const [anchorEl, setAnchorEl] = useState<HTMLDivElement | null>(null)
+  const [open, setOpen] = useState(false)
+  const checkWidth = useRef<HTMLDivElement | null>(null)
+  const currentWidth = checkWidth.current?.clientWidth
+  const colorSegment = useMemo(() => palette[value](), [value])
+
+  const checkDirection = useCallback(() => {
+    const { clientHeight } = document.documentElement
+    const elem = anchorEl?.getBoundingClientRect()
+    if (elem) {
+      const diff = Math.floor(clientHeight - elem.bottom)
+
+      // 300px is the height of <PaletteContainer>
+      if (diff < 300) setDirectionUp(true)
```

Comment:
I think it would be better to use `300` as a constant in this file in that case.

Reply:
Thanks for the feedback, It is changed now.